### ISSUE TITLE
Support for a way to remove a regex101 link that contains sensitive information.

### ISSUE THREAD

**Issue by Zarthus** [firasdib/Regex101#754]
Currently, there's no way to delete a link that contains sensitive information provided you're not logged in (like most users are) and didn't save the deletion link (like most users didn't).

In addition, you might come across your email on a google search because some employee wasn't careful with your data (we get that on IRC more than you'd think.)

Any idiot can come up with a search that gets this easily. Here's one, for instance:
[Google Search: site:regex101.com "@gmail.com"](https://www.google.nl/search?q=site%3Aregex101.com+%22%40gmail.com%22&oq=site%3Aregex101.com+%22%40gmail.com%22&gs_l=serp.3...2049.5463.0.5615.13.12.0.0.0.0.123.604.11j1.12.0....0...1c.1.64.serp..1.0.0.2MFPOcrbTf0)

Ways to solve this:

- Add robots.txt that disallows crawling `r/*/*` (e.g. `r/abcdef/3`) (still leaving libraries public, they start with `library/`)
- Add deletion links to `localStorage`, allowing unregistered users to retrieve it even if they've not saved their deletion links for the duration. So long as they don't clear browser cache.
- Allow long-ish time registered members to flag regexes submitted by no author (unregistered users) for deletion. When a regex receives $x (I suggest three, although this value makes it abusable, it's less abusable than just submitting a bunch of private data for fun.) flags, it gets deleted.

I'm open for better ways to solve this, I'm in agreement with firasdib that a non automated process is a no-go, considering regex101.com is not a professional(e.g. made by a company with income that can afford support employees) website, is completely ad free and maintaining such a thing would be annoying.

**Comment by siuhtk** [comment]
@firasdib: @Zarthus mentioned that he would prod you to check out my issue (#762). Do you maybe have time to look at the personal data issue I posted? I would really appreciate it.

**Comment by Zarthus** [comment]
He mentioned he saw it on irc, if it's not removed I think he just couldn't
be bothered.

**Comment by Zarthus** [comment]
```
(19:01:44) <re101-barb-072> hi , can you delete https://regex101.com/r/<snip>/ and all its child links?
(19:02:02) <re101-barb-072> i was trying to test something and by mistake saved it without account
(19:02:05) <+OnlineCop> When you created it, didn't it give you a delete token?
(19:02:15) <re101-barb-072> i forgot to save the link
(19:02:24) <re101-barb-072> sorry
(19:02:45) <re101-barb-072> created an account now
(19:06:15) <re101-barb-072> thanks much
(19:06:33) <re101-barb-072> in advance
```

**Comment by Zarthus** [comment]
Got an email today from someone who wanted his regex deleted, told him to show support on this issue.

**Comment by lope** [comment]
This issue can be solved very easily by

1. document.cookie = 'cookie1=https://www.regex101.com/delete/abc123 expires=Fri, 3 Aug 2030 20:47:11 UTC; path=/';

2. Place a delete button at the bottom of the page when a regex is saved. The current facility is very fragile. If you accidentally click outside of the div that shows the delete link, the whole div seems to be removed permanently. (not just hidden)

It should not be so difficult to remove information that has been posted.

How can you post information online without providing any realistic way to delete it?

It's highly irresponsible to make it so difficult to delete personal data without providing some kind of warning confirmation when the information is saved.

As I said, even just clicking outside of the div makes the delete code disappear permanently.

How can you punish people by leaving their personal information online when they had no idea how dangerous your site is?

It's insane

**Comment by Zarthus** [comment]
I feel your response is a bit over the top. It's not difficult at all to delete data, and various means are already provided. There just isn't a very good one for users who are not registered.

Regex101 is not meant to store personal data. It goes without saying that you should never, at any point at all, upload personal data without the expectation you cannot delete it. Once it's online on regex101 for even just a few minutes it's not impossible a crawler has already caught your data and stored it. Your frustration partially assumes that it is perfectly okay to store private data online with the expectation you can later revoke it, which is entirely not what you should expect.

What is bad is that you can't really prevent further damage at that point if you haven't stored the deletion link, which is what this issue is about.

**Comment by firasdib** [comment]
Done

**Comment by qoalu** [comment]
i am not able to delete a version of a regex owned by someone else. I was not informed that it would be saved in any way.

**Comment by Doqnach** [comment]
Do you know the other person? If so, contact the person and ask them to delete it.

**Comment by qoalu** [comment]
dont know that person. I found the link on the internet and tested my data against it. Why not implement something similar to codesandbox and the rest, where you can't modify others demos

**Comment by Doqnach** [comment]
@qoalu so let me get the use-case straight:

* You used an existing saved regex, and saved your own data to it as an update?

First of all you should have used the 'fork' feature. Second of all it is unwise to post sensitive data to a public online space (even more so if you are not sure about the effect and persistence). The save button also clearly conveys that it **will** be saved. Nothing will be saved unless you press that button, so you must have had some interaction with it then.

Please create a new issue describing what your use-case is (in this case: "wanting to delete a specific version that you saved to a regex that another user created") since it is different from this issue (that is already closed).

**Comment by qoalu** [comment]
yes.

I see. Usually I can't save something other people created, so I did not expect this behaviour.

There is no save button (or at least I dont see it). There is an "Update regex" function which I did not click, I have a tendency to always "cmd + s" (been burned many times by crashes). And, I assume this caused the saving.

It looks like I am not the only person falling into this trap, and I wonder what I might find there:
![image](https://user-images.githubusercontent.com/1624045/68783578-7ee51180-063b-11ea-8402-cc7bf17e9363.png)

Yes, I am stupid for posting this to a website I don't trust, and have no idea about how it works.

Will make a new issue :)

Thanks for responding
